config/terraform_cloud_workspace_test.go

```diff
@@ -22,23 +25,43 @@ func TestTerraformCloudWorkspaceConfig_Copy(t *testing.T) {
 			"empty",
 			&TerraformCloudWorkspaceConfig{},
 		},
-		{
-			"fully_configured",
-			&TerraformCloudWorkspaceConfig{
-				ExecutionMode:    String("agent"),
-				AgentPoolID:      String("apool-123"),
-				AgentPoolName:    String("test_agent_pool"),
-				TerraformVersion: String("1.0.0"),
-			},
-		},
 	}
-
 	for _, tc := range cases {
 		t.Run(tc.name, func(t *testing.T) {
 			c := tc.c.Copy()
 			assert.Equal(t, tc.c, c)
 		})
 	}
+
+	t.Run("fully_configured", func(t *testing.T) {
+		r := rand.New(rand.NewSource(time.Now().UnixNano()))
+		mode := "test-mode"
+		id := "test-id"
+		name := "test-name"
+		version := "test-version"
+
+		o := &TerraformCloudWorkspaceConfig{
+			ExecutionMode:    &mode,
+			AgentPoolID:      &id,
+			AgentPoolName:    &name,
+			TerraformVersion: &version,
+		}
+		c := o.Copy()
+		assert.Equal(t, o, c)
+
+		// Update original values, check that a deep copy was made
+		mode = fmt.Sprint(r.Intn(999999))
+		assert.NotEqual(t, o.ExecutionMode, c.ExecutionMode)
+
+		id = fmt.Sprint(r.Intn(999999))
+		assert.NotEqual(t, o.AgentPoolID, c.AgentPoolID)
+
+		name = fmt.Sprint(r.Intn(999999))
+		assert.NotEqual(t, o.AgentPoolName, c.AgentPoolName)
+
+		version = fmt.Sprint(r.Intn(999999))
+		assert.NotEqual(t, o, c.TerraformVersion)
```

Comment:
Looks like `o` here might be `o.TerraformVersion`?

Reply:
🤦‍♀️